### ISSUE TITLE
Squiz.Commenting.FunctionCommentThrowTag.WrongNumber not triggered if no throws statement in function body

### ISSUE THREAD

**Issue by smileBeda** [squizlabs/PHP_CodeSniffer#3899]
## Bug Description
`Squiz.Commenting.FunctionCommentThrowTag.WrongNumber` is only triggered if at least one `throw` is present in the function body, and a wrong number of `@throws` docbloc statements is present.
It is not triggered, if a wrong number of `@throws` docbloc statements is present but no `throw` statement is present in the function body.

## Minimal Code Snippet
```
/**
 * Short Description
 *
 * @throws \AnException Comment.
 * @throws \AnotherException Comment.
 */
function check_it() {

	throw new \AnException(
		//do things
	);

}
```

Above will trigger the error.

However, if you do this, the error is not triggered:
```
/**
 * Short Description
 *
 * @throws \AnException Comment.
 * @throws \AnotherException Comment.
 */
function check_it() {

}
```

## Error Code
`Squiz.Commenting.FunctionCommentThrowTag.WrongNumber`

## Environment

PHP 8, WPCS 3.x, all dependencies updated to latest.
No dev.

**Comment by rodrigoprimo** [comment]
This was previously discussed in https://github.com/squizlabs/PHP_CodeSniffer/issues/2683. My understanding from that discussion is that this behavior is by design.

Since PHPCS analyzes files in isolation, it has no way to know if a `@throws` tag is there because of an exception thrown by a called method. So the sniff chose to not report when it can't be sure, to avoid false positives.

I believe this issue can be closed.

**Comment by smileBeda** [comment]
Then it also shouldn’t throw error if there’s at least one throw in the body, this doesn’t compute for me.

Either you check if the count is correct or you don’t.

But then again, it’s been _3 years_ and honestly I’ve moved on, thanks for replying and feel  free to close - although by pure written logic it doesn’t make sense, still.
